sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/ProactiveConnectionManagementTest.java

```diff
@@ -0,0 +1,393 @@
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.azure.cosmos.implementation.TestUtils.mockDiagnosticsClientContext;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+public class ProactiveConnectionManagementTest extends TestSuiteBase {
+
+    private CosmosClientBuilder clientBuilder;
+    private DatabaseAccount databaseAccount;
+    private CosmosAsyncDatabase cosmosAsyncDatabase;
+
+    @BeforeClass(groups = {"multi-region"})
+    public void beforeClass() {
+        clientBuilder = new CosmosClientBuilder()
+                .endpoint(TestConfigurations.HOST)
+                .key(TestConfigurations.MASTER_KEY)
+                .contentResponseOnWriteEnabled(true)
+                .directMode();
+
+        CosmosAsyncClient dummyClient = new CosmosClientBuilder()
+                .endpoint(TestConfigurations.HOST)
+                .key(TestConfigurations.MASTER_KEY)
+                .contentResponseOnWriteEnabled(true)
+                .directMode().buildAsyncClient();
+
+        this.cosmosAsyncDatabase = getSharedCosmosDatabase(dummyClient);
+
+        AsyncDocumentClient asyncDocumentClient = ReflectionUtils.getAsyncDocumentClient(dummyClient);
+        RxDocumentClientImpl rxDocumentClient = (RxDocumentClientImpl) asyncDocumentClient;
+        GlobalEndpointManager globalEndpointManager =
+                ReflectionUtils.getGlobalEndpointManager(rxDocumentClient);
+        this.databaseAccount = globalEndpointManager.getLatestDatabaseAccount();
+
+        safeClose(dummyClient);
+    }
+
+    @Test(groups = {"multi-region"}, dataProvider = "invalidProactiveContainerInitConfigs")
+    public void openConnectionsAndInitCachesWithInvalidCosmosClientConfig(List<String> preferredRegions, int numProactiveConnectionRegions, int numContainers) {
+
+        CosmosAsyncClient clientWithOpenConnections = null;
+
+        List<CosmosAsyncContainer> asyncContainers = new ArrayList<>();
+        List<CosmosContainerIdentity> cosmosContainerIdentities = new ArrayList<>();
+
+        for (int i = 1; i <= numContainers; i++) {
+            String containerId = String.format("id%d", i);
+            cosmosAsyncDatabase.createContainerIfNotExists(containerId, "/mypk").block();
+            asyncContainers.add(cosmosAsyncDatabase.getContainer(containerId));
+            cosmosContainerIdentities.add(new CosmosContainerIdentity(cosmosAsyncDatabase.getId(), containerId));
+        }
+
+        if (numProactiveConnectionRegions > 5) {
+            try {
+                CosmosContainerProactiveInitConfig proactiveContainerInitConfig = new CosmosContainerProactiveInitConfigBuilder(cosmosContainerIdentities)
+                    .setProactiveConnectionRegions(numProactiveConnectionRegions)
+                    .build();
+                fail("Should have thrown exception");
+            } catch (IllegalArgumentException illegalArgEx) {}
```

Comment:
Do we need to verify the right exception info/message is in the exception (here and below) or just verifying getting the exception is ok and standard practice for the SDK tests?

Reply:
exception message can change anytime - and is usually not something we validate - changing exception messages wouldn't be considered breaking  while changing the exception type definitely would. So, this is fine IMO.